executor/executor.go

```diff
@@ -795,16 +795,112 @@ type TableScanExec struct {
 	isVirtualTable     bool
 	virtualTableRows   [][]types.Datum
 	virtualTableCursor int
+
+	// for chunk execution
+	virtualTableChunkList *chunk.List
+	virtualTableChunkIdx  int
 }
 
 // Next implements the Executor interface.
 func (e *TableScanExec) Next(goCtx goctx.Context) (Row, error) {
 	if e.isVirtualTable {
 		return e.nextForInfoSchema()
 	}
+	handle, found, err := e.nextHandle()
+	if err != nil || !found {
+		return nil, errors.Trace(err)
+	}
+	row, err := e.getRow(handle)
+	e.seekHandle = handle + 1
+	return row, errors.Trace(err)
+}
+
+func (e *TableScanExec) nextForInfoSchema() (Row, error) {
+	if e.virtualTableRows == nil {
+		columns := make([]*table.Column, e.schema.Len())
+		for i, v := range e.columns {
+			columns[i] = table.ToColumn(v)
+		}
+		err := e.t.IterRecords(e.ctx, nil, columns, func(h int64, rec []types.Datum, cols []*table.Column) (bool, error) {
+			e.virtualTableRows = append(e.virtualTableRows, rec)
+			return true, nil
+		})
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	if e.virtualTableCursor >= len(e.virtualTableRows) {
+		return nil, nil
+	}
+	row := e.virtualTableRows[e.virtualTableCursor]
+	e.virtualTableCursor++
+	return row, nil
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (e *TableScanExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.isVirtualTable {
+		return errors.Trace(e.nextChunk4InfoSchema(goCtx, chk))
+	}
+	handle, found, err := e.nextHandle()
+	if err != nil || !found {
+		return errors.Trace(err)
+	}
+
+	for chk.NumRows() < e.ctx.GetSessionVars().MaxChunkSize {
+		row, err := e.getRow(handle)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		e.seekHandle = handle + 1
+		for i, col := range e.Schema().Columns {
+			appendDatum2Chunk(chk, &row[i], i, col.RetType)
```

Comment:
consider use `chunk.MutRow`.
1. set Datums to `MutRow`
2. Append `MutRow` to chunk.